### PR TITLE
middleware to redirect education to service.gov.uk

### PR DESCRIPTION
**Pull request by vigneshmsft**
redirect {scheme}://{*}.education.gov.uk/{*} to
{scheme}://{*}.service.gov.uk/{*}

## Context

Followup to #2486 

## Changes proposed in this pull request

middleware to redirect https://*.apply-for-teacher-training.education.gov.uk/ to https://*.apply-for-teacher-training.service.gov.uk/

## Link to Trello card

https://trello.com/c/rIE4NKtK/2414-redirect-educationgovuk-to-servicegovuk

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/middlewares/redirect_to_service_gov_uk_middleware.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  def req_host_is_education_gov_uk(request)
+    request.host.include? 'education.gov.uk'
+  end
+
+  def service_gov_uk_host(request)
+    request.host.sub 'education.gov.uk', 'service.gov.uk'
+  end
+
+  def each(&block) end
```
Comment (duncanjbrown) [inline app/middlewares/redirect_to_service_gov_uk_middleware.rb:23]: Do we need this?

File: config/application.rb [hunk 2]
```diff
@@ -51,6 +52,7 @@ class Application < Rails::Application
 
     config.active_job.queue_adapter = :sidekiq
 
+    config.middleware.insert_after ActionDispatch::RequestId, RedirectToServiceGovUkMiddleware
```
Comment (duncanjbrown) [inline config/application.rb:55]: Is this the earliest it could possibly go?

File: app/middlewares/redirect_to_service_gov_uk_middleware.rb (hunk 1 above)
Comment (vigneshmsft) [reply app/middlewares/redirect_to_service_gov_uk_middleware.rb:23]: yes, else it throws an method not found error

File: config/application.rb (hunk 2 above)
Comment (vigneshmsft) [reply config/application.rb:55]: > Looks good — keen to see this on QA.
> 
> Could we perhaps only insert the middleware on qa and staging as

Deployed it to QA last night for testing, can do a deploy again if you'd like to take a look.

Comment (vigneshmsft) [reply config/application.rb:55]: @duncanjbrown deployed it to QA https://qa.apply-for-teacher-training.education.gov.uk/ (should take you to service.gov.uk)
have also inserted the middleware after `ActionDispatch::HostAuthorization` will try inserting before next

Comment (vigneshmsft) [reply config/application.rb:55]: I think its better to leave it after https://edgeapi.rubyonrails.org/classes/ActionDispatch/HostAuthorization.html ?

Comment (vigneshmsft) [reply config/application.rb]: configured it after `ActionDispatch::HostAuthorization` as it protects against unauthorised hosts.

**Pull request merged**